script/Deploy.s.sol

```diff
@@ -36,10 +36,20 @@ contract Deploy is Script {
         // refer to https://docs.openzeppelin.com/upgrades-plugins/foundry-upgrades#upgrade_a_proxy_or_beacon for more details
         Options memory opts;
         opts.referenceContract = "BatchInbox.sol:BatchInbox";
-        Upgrades.upgradeProxy(proxyAddress, "BatchInboxV2.sol:BatchInboxV2", "", opts);
+
+        // Upgrades.upgradeProxy(proxyAddress, "BatchInboxV2.sol:BatchInboxV2", "", opts);
+
+        // prepareUpgrade does not upgrade the proxy, it only prepares the new implementation address
+        // and returns the address of the new implementation.
+        // You need to call the upgradeTo function on the proxy manually after this.
+        // This is useful for multisig wallets or other scenarios where you want to prepare the upgrade
+        // and then execute it later.
+        // See https://github.com/OpenZeppelin/openzeppelin-upgrades/issues/946#issuecomment-1861378609
+        address newImpl = Upgrades.prepareUpgrade("BatchInboxV2.sol:BatchInboxV2", opts);
```

Comment:
`BatchInboxV2.sol` is another file, which will be hard to view the diff, maybe we can just modify `BatchInbox.sol` in place for future updates?

Reply:
Also, opts.referenceContract and the new deployment file must be different; otherwise, the plugin will throw a validation error.